com.hybris.hyeclipse.commons/src/com/hybris/hyeclipse/commons/utils/EclipseFileUtils.java

```diff
@@ -0,0 +1,148 @@
+package com.hybris.hyeclipse.commons.utils;
+
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+
+import org.apache.commons.io.IOUtils;
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.Platform;
+import org.eclipse.jface.text.TextSelection;
+import org.eclipse.jface.viewers.ISelection;
+import org.eclipse.jface.viewers.IStructuredSelection;
+import org.eclipse.ui.IEditorPart;
+import org.eclipse.ui.IWorkbenchWindow;
+import org.eclipse.ui.PlatformUI;
+import org.eclipse.ui.texteditor.ITextEditor;
+
+/**
+ * Utility class to work with files in eclipse.
+ */
+public final class EclipseFileUtils {
+
+	/**
+	 * Private in order to avoid class initialization
+	 */
+	private EclipseFileUtils() {
+		/* Intentionally empty */ }
+
+	/**
+	 * @return workbench active window
+	 */
+	public static IWorkbenchWindow getActiveWorkbenchWindow() {
+		return PlatformUI.getWorkbench().getActiveWorkbenchWindow();
+	}
+	
+	/**
+	 * @return active editor file
+	 */
+	public static IFile getActiveEditorFile() {
+		return (IFile) getActiveWorkbenchWindow().getActivePage().getActiveEditor().getEditorInput().getAdapter(IFile.class);
+	}
+	
+	/**
+	 * Returns selected file
+	 * 
+	 * @param selection
+	 *            current selection
+	 * @return selected {@link IFile}
+	 */
+	public static IFile getSelectedFile(final ISelection selection) {
+		IFile file = null;
+		
+		if (selection instanceof IStructuredSelection) {
+			final IStructuredSelection ssel = (IStructuredSelection) selection;
+			final Object obj = ssel.getFirstElement();
+			file = (IFile) Platform.getAdapterManager().getAdapter(obj, IFile.class);
+		} else if (selection instanceof TextSelection) {
+			file = getActiveEditorFile();
+		}
+		return file;
+	}
+
+	/**
+	 * Return set of selected file(s)
+	 * 
+	 * @param selection
+	 *            current selection
+	 * @return set of selected files
+	 */
+	@SuppressWarnings("unchecked")
+	public static Set<IFile> getSelectedFiles(final ISelection selection) {
+		final Set<IFile> files = new HashSet<>();
+
+		if (selection instanceof IStructuredSelection) {
+			final IStructuredSelection structuredSelection = (IStructuredSelection) selection;
+			structuredSelection.toList().forEach(
```

Comment:
Maybe we can use stream ? We can also extract the Platform.getAdapterManager().getAdapter to separate method ? Since form contex I dont know what it does. 
```
structuredSelection.toList().stream()
.map(file -> Platform.getAdapterManager().getAdapter(file, IFile.class))) 
.map(IFile.class::cast)
.forEach(files::add)
```